swift-sdk/IterableAPIInternal.swift

```diff
@@ -872,6 +908,10 @@ import Foundation
         // get email and userId from UserDefaults if present
         retrieveEmailAndUserId()
         
+        if config.autoPushRegistration == true && isEitherUserIdOrEmailSet() {
+            notificationStateProvider.registerForRemoteNotifications()
```

Comment:
This may be called on a background thread, maybe `SystemNotificationStateProvider` should wrap its call in `DispatchQueue.main.async` so we don't have to do that in the caller?